Add built-in monitoring (#9774)
pingcap/docs: tidb-cloud/monitor-built-in-monitoring.md

---
title: TiDB Cloud Built-in Monitoring
summary: Learn how to view TiDB Cloud built-in monitoring metrics and understand the meanings of these metrics.
---

# TiDB Cloud Built-in Monitoring

TiDB Cloud collects and displays a full standard metrics of your cluster in TiDB Cloud console. By viewing these metrics, you can easily identify performance issues and determine whether your current database deployment meets your requirements.

> **Note:**
>
> For [Developer Tier clusters](/tidb-cloud/select-cluster-tier.md#developer-tier), the Monitoring page is unavailable. You can upgrade to [Dedicated Tier clusters](/tidb-cloud/select-cluster-tier.md#dedicated-tier) to use it.

## How to use feature 

To view these metrics, take the following steps:

> 1. Navigate to the **Diagnostics** tab of a cluster.

Reviewer: ```suggestion
1. Navigate to the **Diagnosis** tab of a cluster.
```

> 1. Navigate to the **Diagnostics** tab of a cluster.
> 
> 2. Click the **Monitoring** tab.

Reviewer 2: ```suggestion
1. In the TiDB Cloud console, navigate to the **Active Clusters** page for your project, and then click the name of a cluster to go to its overview page.
2. Click the **Diagnostics** tab.
3. Click the **Monitoring** tab.
```

## Metrics Content

The following sections illustrate the metrics on the Monitoring page.

### Database Time

| Metric name  | Labels | Description                                   |
| :------------| :------| :-------------------------------------------- |
| Database Time | database time | Total database time per second. |
| Database Time by SQL types | Select, Insert, Use, etc. | Database time consumed by each type of SQL statements per second. |
| Database Time by SQL Phase | get token, parse, compile, execute | Database time consumed in four SQL processing phases: get token, parse, compile, and execute. The SQL execution phase is in green and other phases are in red on general. If non-green areas take a large proportion, it means most database time is consumed by other phases than the execution phase and further cause analysis is required. |

> | SQL Execute Time Overview | tso_wait, Get, Cop, Commit, etc. | Green metrics stand for common KV write requests (such as prewrite and commit), blue metrics stand for common read requests, and metrics in other colors stand for unexpected situations which you need to pay attention to. For example, pessimistic lock KV requests are marked red and TSO waiting is marked dark brown.If non-blue or non-green areas take a large proportion, it means there is a bottleneck during SQL execution. For example, if serious lock conflicts occur, the red area will take a large proportion. If excessive time is consumed in waiting TSO, the dark brown area will take a large proportion. |

Reviewer: ```suggestion
| SQL Execute Time Overview | tso_wait, Get, Cop, Commit, etc. | Green metrics stand for common KV write requests (such as prewrite and commit), blue metrics stand for common read requests, and metrics in other colors stand for unexpected situations which you need to pay attention to. For example, pessimistic lock KV requests are marked red and TSO waiting is marked dark brown. If non-blue or non-green areas take a large proportion, it means there is a bottleneck during SQL execution. For example, if serious lock conflicts occur, the red area will take a large proportion. If excessive time is consumed in waiting TSO, the dark brown area will take a large proportion. |
```

| Average / P99 Apply Log Duration | avg, 99 | The average time or P99 duration consumed by Raft to apply logs. |

### Server

| Metric name  | Labels | Description                                   |
| :------------| :------| :-------------------------------------------- |
| TiDB Uptime | instances | The runtime of each TiDB instance since last restart. |
| TiDB CPU Usage | instances | The statistics of CPU usage of each TiDB instance. |
| TiDB Memory Usage | instances | The memory usage statistics of each TiDB instance. |
| TiKV Uptime | instances | The runtime of each TiKV instance since last restart. |
| TiKV CPU Usage | instances | The statistics of CPU usage of each TiKV instance. |
| TiKV Memory Usage | instances | The memory usage statistics of each TiKV instance. |
| TiKV IO MBps | instances-write, instances-read | The total bytes of read and write in each TiKV instance. |
| TiKV Storage Usage | instances | The storage size per TiKV instance. |
| TiFlash Uptime | instances | The runtime of each TiFlash instance since last restart. |
| TiFlash CPU Usage | instances | The statistics of CPU usage of each TiFlash instance. |

> | TiFlash Memory Usage | instances | The memory usage statistics of each TiFlash instance. |

Reviewer 2: ```suggestion
| TiFlash Memory  | instances | The memory usage statistics of each TiFlash instance. |
```

| TiFlash IO MBps | instances-write, instances-read | The total bytes of read and write in each TiFlash instance. |
| TiFlash Storage Usage | instances | The storage size per TiFlash instance. |

## FAQ

### 1. Why are some panels empty on this page？

If there is no value in the panel, there may be several scenarios.
- The cluster's workload does not trigger this metric. For example, the failed query metric is always empty in the case of no failed queries.

> - The cluster version is low, please upgrade to the latest version of TiDB to see these metrics.

Reviewer 2: ```suggestion
- The cluster version is low. You need to upgrade it to the latest version of TiDB to see these metrics.
```

> - If all the above scenarios are excluded, please contact technical support.

Reviewer 2: ```suggestion

If all these reasons are excluded, you can contact the [PingCAP support team](/tidb-cloud/tidb-cloud-support.md) for troubleshooting.
```
